runtime/src/bank.rs

```diff
@@ -1004,6 +1005,7 @@ impl Bank {
         bank.transaction_account_lock_limit = runtime_config.transaction_account_lock_limit;
         bank.transaction_debug_keys = debug_keys;
         bank.cluster_type = Some(genesis_config.cluster_type);
+        bank.feature_set = Arc::new(feature_set.unwrap_or_default());
```

Comment:
This code path shouldn't be available in production. It would better be gated behind a `#[cfg(feature = "dev-context-only-utils")]` attribute.